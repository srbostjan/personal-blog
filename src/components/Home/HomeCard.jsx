import React from 'react';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';

function HomeCard({ to, name }) {
  const url = to.startsWith('http');
  const navigate = useHistory();
  const icon = () => {
    if (name.toLowerCase() === 'twitter') return 'fab fa-twitter';
    if (name.toLowerCase() === 'github') return 'fab fa-github';
    return 'fas fa-arrow-circle-right';
  };
  const handleClick = () => {
    if (url) {
      window.open(to, '_blank');
    } else {
      navigate.push(`/${to}`);
    }
  };
  return (
    <button type="button" onClick={handleClick} className="home-card">
      <div className="circle-animation" />
      <p>
        { name }
        <i className={icon()} />
      </p>
    </button>
  );
}

HomeCard.propTypes = {
  name: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
};

export default HomeCard;
