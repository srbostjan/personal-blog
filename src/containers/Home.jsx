import React from 'react';
import HomeCard from '@components/Home/HomeCard';
import DownArrow from '@components/DownArrow';

function Home() {
  const download = 'https://drive.google.com/uc?id=1kV_x1dJ1R6sVWuXdtWUhK2iTg-DAgV5f&export=download';
  return (
    <div className="main-home">
      <section className="home-left">
        <div className="content">
          <h1>Frontend Developer</h1>
          <p>UI:UX enthusiasth</p>
          <p>Clean code</p>
          <button type="button"><a className="cv-download-link" href={download} download="Frontend_SebastianRiveraOcampo_CV">Download CV</a></button>
        </div>
      </section>
      <section className="home-right" id="home-right">
        <HomeCard name="Biography" to="biography" />
        <HomeCard name="Articles" to="articles" />
        <HomeCard name="Twitter" to="https://twitter.com/srbostjan" />
        <HomeCard name="Github" to="https://github.com/srbostjan" />
      </section>
      <DownArrow />
    </div>
  );
}

export default Home;
