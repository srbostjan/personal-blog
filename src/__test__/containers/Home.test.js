import React from 'react';
import { mount, shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Home from '../../containers/Home';

describe('<Home />', () => {
  const component = shallow(<Home />);
  test('Render componente Home', () => {
    expect(component.length).toEqual(1);
  });
  test('Render de titulo', () => {
    expect(component.find('h1').text()).toEqual('Frontend Developer');
  })
});
describe('Home Snapshot', () => {
  test('Home Snapshot', () => {
    const component = create(<Home />);
    expect(component.toJSON()).toMatchSnapshot();
  });
})