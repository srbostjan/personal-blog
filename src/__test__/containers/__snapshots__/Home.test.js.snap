// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot Home Snapshot 1`] = `
<div
  className="main-home"
>
  <section
    className="home-left"
  >
    <div
      className="content"
    >
      <h1>
        Frontend Developer
      </h1>
      <p>
        UI:UX enthusiasth
      </p>
      <p>
        Clean code
      </p>
      <button
        onClick={[Function]}
        type="button"
      >
        <h3>
          Download CV
        </h3>
      </button>
    </div>
  </section>
  <section
    className="home-right"
    id="home-right"
  >
    <button
      className="home-card"
      onClick={[Function]}
      type="button"
    >
      <div
        className="circle-animation"
      />
      <p>
        Biography
        <i
          className="fas fa-arrow-circle-right"
        />
      </p>
    </button>
    <button
      className="home-card"
      onClick={[Function]}
      type="button"
    >
      <div
        className="circle-animation"
      />
      <p>
        Articles
        <i
          className="fas fa-arrow-circle-right"
        />
      </p>
    </button>
    <button
      className="home-card"
      onClick={[Function]}
      type="button"
    >
      <div
        className="circle-animation"
      />
      <p>
        Twitter
        <i
          className="fab fa-twitter"
        />
      </p>
    </button>
    <button
      className="home-card"
      onClick={[Function]}
      type="button"
    >
      <div
        className="circle-animation"
      />
      <p>
        Github
        <i
          className="fab fa-github"
        />
      </p>
    </button>
  </section>
  <button
    onClick={[Function]}
    type="button"
  >
    <i
      className="fas fa-arrow-circle-right down-arrow"
    />
  </button>
</div>
`;
